master/pkg/tasks/task_gc.go

```diff
@@ -37,10 +37,17 @@ func (g GCCkptSpec) ToTaskSpec() TaskSpec {
 	// Set Environment.
 	// Keep only the EnvironmentVariables provided by the experiment's config.
 	envVars := g.LegacyConfig.Environment.EnvironmentVariables()
+
+	podSpec := g.LegacyConfig.Environment.PodSpec()
+	if g.Base.TaskContainerDefaults.CheckpointGCPodSpec != nil {
+		gcPodSpec := (*expconf.PodSpec)(g.Base.TaskContainerDefaults.CheckpointGCPodSpec)
+		podSpec = schemas.Merge(podSpec, gcPodSpec)
```

Comment:
I don't think we should merge the pod spec from the experiment here in case it has node affinities that we won't want to add

Reply:
Noted! 
Replaced the code with overwriting the legacy PodSpec value